website/homepage/src/demos/restart_demos.py

```diff
@@ -0,0 +1,10 @@
+from upload_demos import demos_by_category, upload_demo_to_space, AUTH_TOKEN, gradio_version
+from gradio.networking import url_ok
+
+
+for category in demos_by_category:
+    for demo in category["demos"]:
+        space_id = "gradio/" + demo["dir"]
+        if not url_ok(f"https://hf.space/embed/{space_id}/+"):
```

Comment:
note that those URLs will change soon-ish :) 

(but when they change, they will be then considered "public API" ie. they won't change again after that)

Reply:
Thanks for clarifying @julien-c, will make sure to change this when the urls change. 

Do you know if there's a way in the api to programmatically check if the space has a build error? Instead of the way I'm doing it now which is checking if the url returns 200

Reply:
yes there's a way 

Stage should be exposed in `SpaceInfo` in `huggingface_hub`, for instance